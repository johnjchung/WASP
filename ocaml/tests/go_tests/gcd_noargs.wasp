Func int gcd(int a, int b)
{
  while (a != b) {
    if (a > b) {a = a-b;
    }
    else {b = b-a;}
    }
    return a;
}
Endpoint gcdtest(GET:) {
  int a = gcd(3, 6);
}
