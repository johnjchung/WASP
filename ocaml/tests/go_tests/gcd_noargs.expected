package main

import (
	"github.com/gin-gonic/gin"
	"math"
	"os"
)

func int_conv(anything interface{}) int {
	switch v := anything.(type) {
		case int:
			return v
		case float64, float32:
			var float_var = v.(float64)
			return int(math.Trunc(float_var))
		default:
			return 0
	}
	return 0
}

func float_conv(anything interface{}) float64 {
 	 switch v := anything.(type) {
		case int:
            return float64(v)
		case float64:
			return v
		default:
			return 0
	}
	return 0.0
}

type gcdtest_struct struct {
	A int `db:"a" json:"a"`
}

func main() {
	router := gin.Default()
	router.GET("/gcdtest", Get_gcdtest)
	router.Run(":"+os.Getenv("PORT"))
}

func gcd(a int, b int) int {
	for a != b {
	if bool(a > b) {
	a = int_conv(a) - int_conv(b)
	} else {
	b = int_conv(b) - int_conv(a)
	}
	}
	return a
}

func gcdtest_struct_wrapper() gcdtest_struct {
	return gcdtest_struct{A: gcd(3, 6)}
}

func Get_gcdtest(c *gin.Context) {
	gcdtest_struct := gcdtest_struct_wrapper()
	c.JSON(200, gcdtest_struct)
}
